influxdbconnector/src/main/java/org/openbase/bco/app/influxdbconnector/InfluxDbconnectorApp.java

```diff
@@ -22,48 +22,255 @@
 import org.openbase.jul.exception.printer.ExceptionPrinter;
+import org.openbase.jul.pattern.Observer;
 import org.openbase.jul.schedule.GlobalCachedExecutorService;
 import org.openbase.type.domotic.action.ActionDescriptionType.ActionDescription;
+import org.openbase.type.domotic.service.ServiceDescriptionType;
+import org.openbase.type.domotic.service.ServiceTemplateType;
+import org.openbase.type.domotic.service.ServiceTempusTypeType;
 import org.openbase.type.domotic.state.ActivationStateType.ActivationState;
+import org.openbase.type.domotic.unit.UnitConfigType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
 
-public class InfluxDbconnectorApp extends AbstractAppController  {
+import static org.openbase.bco.dal.lib.layer.service.Services.resolveStateValue;
+
+public class InfluxDbconnectorApp extends AbstractAppController {
 
     private Logger logger = LoggerFactory.getLogger(getClass());
 
     private Future task;
+    private String databaseUrl;
+    private String databaseName;
+    private InfluxDB influxDB;
+    private Integer batchTime;
+    private Integer batchLimit;
+    private CustomUnitPool customUnitPool;
+    private Observer<ServiceStateProvider<Message>, Message> unitStateObserver;
+    private static final Integer READ_TIMEOUT = 60;
+    private static final Integer WRITE_TIMEOUT = 60;
+    private static final Integer CONNECT_TIMOUT = 40;
+
 
     public InfluxDbconnectorApp() throws InstantiationException {
 
+
+
     }
 
     @Override
-    protected ActionDescription execute(ActivationState activationState) throws CouldNotPerformException, InterruptedException {
-        logger.info("execute influx db connector");
+    protected ActionDescription execute(ActivationState activationState) throws CouldNotPerformException {
+        try {
+            databaseUrl = generateVariablePool().getValue("INFLUXDB_URL");
+            databaseName = generateVariablePool().getValue("INFLUXDB_NAME");
+            batchTime = Integer.valueOf(generateVariablePool().getValue("INFLUXDB_BATCH_TIME"));
+            batchLimit = Integer.valueOf(generateVariablePool().getValue("INFLUXDB_BATCH_LIMIT"));
+        } catch (NotAvailableException ex) {
+            ExceptionPrinter.printHistory(ex, logger);
+        }
+
+
+
         task = GlobalCachedExecutorService.submit(() -> {
-            while (!task.isCancelled()) {
-                try {
-                    logger.info("connect to db at: "+ generateVariablePool().getValue("INFLUXDB_URL" ));
-                    Thread.sleep(5000);
-                } catch (InterruptedException | CouldNotPerformException ex) {
-                    ExceptionPrinter.printHistory(ex, logger);
+            logger.info("execute influx db connector");
+            try {
+                if (initiateDatabase()) {
```

Comment:
currently the task would just finish if the influx db is not available during the app startup right?
We definitely need some auto retry. That's what this task is perfectly for.
So in case the initialization fails, wait some time and try it again. (be aware of the exception handling mentioned related to the `initiateDatabase()` method.